pkg/cmd/cli/cmd/examples.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+const examplesLong = `Displays one big list of examples for most commands available in the OpenShift client tools.
```

Comment:
The first purpose was to be the basis for generating the asciidoc document, but when writing the generator script it seemed useful on its own to end-users too. Examples tend to expose all important paths of a given command, so by having a single place to list them all you have a full view of everything the client can do, by examples. Kind of a conceptual command similar to `oc types`. Also, there's --output and --template which can be used to generate different things - it's what I use to generate the asciidoc. ;)

Sounds different to you guys?

----- Original Message -----
> > @@ -0,0 +1,81 @@
> > +package cmd
> > +
> > +import (
> > +	"fmt"
> > +	"io"
> > +
> > +	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
> > +	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > +	"github.com/spf13/cobra"
> > +)
> > +
> > +const examplesLong = `Displays one big list of examples for most commands
> > available in the OpenShift client tools.
> 
> I didn't get an answer for why I want this command as a user if I already
> have the individual commands.
> 
> > On Jun 25, 2015, at 12:17 PM, Jordan Liggitt <notifications@github.com>
> > wrote:
> > 
> > In pkg/cmd/cli/cmd/examples.go:
> > 
> > > @@ -0,0 +1,81 @@
> > > +package cmd
> > > +
> > > +import (
> > > +	"fmt"
> > > +	"io"
> > > +
> > > +	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
> > > +	kcmdutil
> > > "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
> > > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > > +	"github.com/spf13/cobra"
> > > +)
> > > +
> > > +const examplesLong = `Displays one big list of examples for most
> > > commands available in the OpenShift client tools.
> > "Displays examples for most commands in the OpenShift client tools."
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> > 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/3325/files#r33296902